lib/sqlalchemy/sql/schema.py

```diff
@@ -3767,15 +3800,31 @@ def _col_expressions(
             return [
                 parent.c[col] if isinstance(col, str) else col
                 for col in self._pending_colargs
+                if col not in parent.periods
             ]
 
+    def _period_expressions(
```

Comment:
I would assume this logic is needed by the Compiler and that's it.  I would make this classmethod on Period itself:

```py
@classmethod
def _periods_for_pk_constraint(cls, pk_constraint):
    return [p for p in pk_constraint.table.periods if p.start.primary_key or p.end.primary_key]
```